src/common/actions/repositories.js

```diff
@@ -53,8 +54,124 @@ export function fetchUserRepositories(pageNumber) {
 
 export function fetchRepositoriesError(error) {
   return {
-    type: FETCH_REPOSITORIES_ERROR,
+    type: REPOSITORIES_FAILURE,
     payload: error,
     error: true
   };
 }
+
+// XXX may need to split out the fetch from the success and failure dispatch
+// to ensure than setRepositories happens in sync with refreshed user snaps
+export function fetchUserRepositoriesAndSnaps(owner) {
+  return (dispatch) => {
+    return Promise.all([
+      dispatch(fetchUserRepositories()),
+      dispatch(fetchUserSnaps(owner))
+    ]);
+  };
+}
+
+// XXX move to repository actions?
+// could be plural or singular, could be in one module or two, might be clearer in one for now
+export const REPOSITORY_TOGGLE_SELECT = 'REPOSITORY_TOGGLE_SELECT';
+
+export function toggleRepositorySelection(id) {
+  return {
+    type: REPOSITORY_TOGGLE_SELECT,
+    payload: {
+      id
+    }
+  };
+}
+
+export const REPOSITORY_BUILD = 'REPOSITORY_BUILD';
+export const REPOSITORY_SUCCESS = 'REPOSITORY_SUCCESS';
+export const REPOSITORY_FAILURE = 'REPOSITORY_FAILURE';
+export const REPOSITORY_RESET = 'REPOSITORY_RESET';
+
+export function buildRepositories(repositories) {
+  return (dispatch) => {
+    const promises = repositories.map(
+      (repository) => {
+        return dispatch(buildRepository(repository));
+      }
+    );
+    return Promise.all(promises).then(() => dispatch(push('/dashboard')));
+  };
+}
+
+export function buildRepository(repository) {
+  const { id, url, owner, name } = repository;
+
+  return async (dispatch) => {
+    dispatch({
+      type: REPOSITORY_BUILD,
+      payload: {
+        id
+      }
+    });
+
+    try {
+      await createWebhook(owner, name);
+      const response = await fetch(`${BASE_URL}/api/launchpad/snaps`, {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({ repository_url: url }),
+        credentials: 'same-origin'
+      });
+
+      await checkStatus(response);
+      dispatch(buildRepositorySuccess(id));
+    } catch (error) {
+      dispatch(buildRepositoryError(id, error));
+      return Promise.reject(error);
```

Comment:
> "In this case [the] ... error was handled"

What I meant is that perhaps we shouldn't be throwing an error at all:

> The general rule is that a function may deliver operational errors synchronously (e.g., by throwing) or asynchronously (by passing them to a callback or emitting error on an EventEmitter), but it should not do both. 

https://www.joyent.com/node-js/production/design/errors

But since the error is based back via await, it should simply be part of the response?